lib/VM/JSLib/Math.cpp

```diff
@@ -205,18 +207,45 @@ CallResult<HermesValue> mathPow(void *, Runtime &runtime, NativeArgs args) {
   return HermesValue::encodeUntrustedNumberValue(expOp(x, y));
 }
 
+// Help defines for random
+static constexpr int RIGHT12 = 12;
+static constexpr int SECONDS_TO_SUBTLE = 1000000;
+static constexpr int RIGHT27 = 27;
+static constexpr int LEFT25 = 25;
+static constexpr uint64_t GET_MULTIPLY = 0x2545F4914F6CDD1D;
+// Exponent bits for double value between [1.0, 2.0)
+static constexpr uint64_t EXPONENTBITS_RANGE_IN_ONE_AND_TWO = 0x3FF0000000000000;
+
+static double ToDouble(uint64_t state) {
+  uint64_t random = (state >> RIGHT12) | EXPONENTBITS_RANGE_IN_ONE_AND_TWO;
+  return llvh::bit_cast<double>(random) - 1;
+}
+
+static uint64_t XorShift64(uint64_t *pVal) {
+  uint64_t x = *pVal;
+  x ^= x >> RIGHT12;
+  x ^= x << LEFT25;
+  x ^= x >> RIGHT27;
+  *pVal = x;
+  return x * GET_MULTIPLY;
+}
+
 // ES5.1 15.8.2.14
 // Returns a Hermes-encoded pseudo-random number uniformly chosen from [0, 1)
 CallResult<HermesValue> mathRandom(void *, Runtime &runtime, NativeArgs) {
   RuntimeCommonStorage *storage = runtime.getCommonStorage();
-  if (!storage->randomEngineSeeded_) {
-    std::minstd_rand::result_type seed;
-    seed = std::random_device()();
-    storage->randomEngine_.seed(seed);
-    storage->randomEngineSeeded_ = true;
+  if (!storage->randomEngineInited) {
+    struct timeval tv;
+    gettimeofday(&tv, NULL);
+    storage->randomState_ = static_cast<uint64_t>((tv.tv_sec * SECONDS_TO_SUBTLE) + tv.tv_usec);
```

Comment:
I think this is problematic because 
- clock resolution maybe more coarse grained than microseconds. e.g. on Linux it is usually 10us. 
- Even if there is perfect microsecond granularity, you'll start getting collisions if thousands of uuids are generated every second across many devices since many of them will get the same read of microsecond. In general, even though the "microseconds since epoch" is a large value it has very strong temporal locality in the sense that at any point in time all values will be tightly clustered.
- I also feel not so warm about rolling a custom RNG where there are so many available in the standard library specially, without running any of the standard randomness test suites on it.

I'd recommend using `std::random_device()()` to generate multiple 32-bit numbers and concatenating those to form a 64 bit seed or better yet, use a `std::seed_seq()` with enough members to initialize a Mersenne twister to get a very long period.

Reply:
OK, thanks, will try it

Reply:
I have changed the implement